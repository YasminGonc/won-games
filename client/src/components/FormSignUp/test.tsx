import { renderWithTheme } from '../../utils/tests/helpers'
import { screen } from '@testing-library/react'

import { FormSignUp } from '.'

describe('<FormSignUp />', () => {
  it('should render the form', () => {
    const { container } = renderWithTheme(<FormSignUp />)

    expect(screen.getByPlaceholderText('Name')).toBeInTheDocument()
    expect(screen.getByPlaceholderText('Email')).toBeInTheDocument()
    expect(screen.getByPlaceholderText('Password')).toBeInTheDocument()
    expect(screen.getByPlaceholderText('Confirm password')).toBeInTheDocument()
    expect(
      screen.getByRole('button', { name: /Sign up now/i })
    ).toBeInTheDocument()

    expect(container.parentElement).toMatchSnapshot()
  })

  it('should render the text and link to sign in', () => {
    renderWithTheme(<FormSignUp />)

    expect(screen.getByText('Already have an account?')).toBeInTheDocument()
    expect(screen.getByRole('link', { name: /sign in/i })).toBeInTheDocument()
  })
})
