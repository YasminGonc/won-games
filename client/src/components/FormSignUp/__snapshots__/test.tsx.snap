// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormSignUp /> should render the form 1`] = `
.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.8rem;
  background-color: #EAEAEA;
  border-radius: 0.4rem;
  padding: 0.8rem;
}

.c3:focus-within {
  box-shadow: 0 0 0.5rem #F231A5;
}

.c3 svg {
  height: 24px;
  width: 24px;
  color: #8F8F8F;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
}

.c4 {
  color: #2E2F42;
  font-family: 'Poppins',-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
  background-color: transparent;
  border: 0;
  outline: none;
  width: 100%;
}

.c4::-webkit-input-placeholder {
  font-family: 'Poppins',-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.4rem;
}

.c4::-moz-placeholder {
  font-family: 'Poppins',-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.4rem;
}

.c4:-ms-input-placeholder {
  font-family: 'Poppins',-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.4rem;
}

.c4::placeholder {
  font-family: 'Poppins',-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.4rem;
}

.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 0.7rem;
  background: linear-gradient( 178.59deg, #ff5f5f -14.51%, #f062c0 102.86%, #f23131 102.86% );
  border: none;
  border-radius: 0.4rem;
  padding: 0.8rem;
  cursor: pointer;
  color: #FAFAFA;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
  width: 100%;
}

.c6:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 100%);
}

.c0 .c1 {
  margin: 0.8rem 0;
}

.c0 .c5 {
  margin: 3.2rem auto 1.6rem;
}

.c7 {
  color: #030517;
  font-size: 1.4rem;
  text-align: center;
}

.c7 a {
  color: #3CD3C1;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: 0ch.1rem solid #3CD3C1;
  -webkit-transition: color,border,0.1s ease-in-out;
  transition: color,border,0.1s ease-in-out;
  margin-left: 0.2rem;
}

.c7 a:hover {
  color: #29b3a3;
  border-bottom: 0.1rem solid #29b3a3;
}

<body>
  <div>
    <div
      class="c0"
    >
      <form>
        <div
          class="c1 c2"
        >
          <div
            class="c3"
          >
            <svg
              fill="currentColor"
              height="1em"
              viewBox="0 0 256 256"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <circle
                cx="128"
                cy="128"
                fill="none"
                r="96"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
              <circle
                cx="128"
                cy="120"
                fill="none"
                r="40"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
              <path
                d="M63.8,199.4a72,72,0,0,1,128.4,0"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
            </svg>
            <input
              aria-label="Name input"
              class="c4"
              id="Name"
              name="name"
              placeholder="Name"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="c1 c2"
        >
          <div
            class="c3"
          >
            <svg
              fill="currentColor"
              height="1em"
              viewBox="0 0 256 256"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <path
                d="M32,56H224a0,0,0,0,1,0,0V192a8,8,0,0,1-8,8H40a8,8,0,0,1-8-8V56A0,0,0,0,1,32,56Z"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
              <polyline
                fill="none"
                points="224 56 128 144 32 56"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
            </svg>
            <input
              aria-label="Email input"
              class="c4"
              id="Email"
              name="email"
              placeholder="Email"
              type="email"
              value=""
            />
          </div>
        </div>
        <div
          class="c1 c2"
        >
          <div
            class="c3"
          >
            <svg
              fill="currentColor"
              height="1em"
              viewBox="0 0 256 256"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <rect
                fill="none"
                height="128"
                rx="8"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
                width="176"
                x="40"
                y="88"
              />
              <path
                d="M92,88V52a36,36,0,0,1,72,0V88"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
            </svg>
            <input
              aria-label="Password input"
              class="c4"
              id="Password"
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
        </div>
        <div
          class="c1 c2"
        >
          <div
            class="c3"
          >
            <svg
              fill="currentColor"
              height="1em"
              viewBox="0 0 256 256"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <rect
                fill="none"
                height="128"
                rx="8"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
                width="176"
                x="40"
                y="88"
              />
              <path
                d="M92,88V52a36,36,0,0,1,72,0V88"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
            </svg>
            <input
              aria-label="Confirm password input"
              class="c4"
              id="Confirm password"
              name="confirm password"
              placeholder="Confirm password"
              type="password"
              value=""
            />
          </div>
        </div>
        <button
          class="c5 c6"
        >
          Sign up now
        </button>
        <div
          class="c7"
        >
          Already have an account?
          <a
            href="/SignIn"
          >
            Sign in
          </a>
        </div>
      </form>
    </div>
  </div>
</body>
`;
